### PR TITLE
vimeo: use api to get video info, direct download if possible

### PR DESCRIPTION
**Pull request by dumbmoron**

### CODE REVIEW

File: src/modules/processing/services/vimeo.js [hunk 1]
```diff
@@ -84,15 +122,48 @@ export default async function(obj) {
...
+    if (!quality || obj.isAudioOnly) quality = 9000;
+
+    const info = await requestApiInfo(obj.id, obj.password);
+    let response;
+
+    if (obj.isAudioOnly) {
+        response = await getHLS(info.config_url, { ...obj, quality });
+    }
+
+    if (!response) response = await getDirectLink(info, quality);
```
Comment (wukko) [inline src/modules/processing/services/vimeo.js:145]: `getDirectLink()` is not an async function

**Review by wukko — APPROVED** [review]
lgtm

**Review by wukko — APPROVED** [review]
lgtm 2

**Comment by aquarime** [comment]
I just discovered the same issue and came here to check it out.
Impossible to download https://vimeo.com/698459270
Cobalt opens the download to `https://vod-progressive.akamaized.net/exp=1722439422~acl=%2Fvimeo-transcode-storage-prod-us-west1-h264-1080p%2F01%2F4691%2F27%2F698459270%2F3230977345.mp4~hmac=b20e042c1754e392e617a4b80dacd21624464860e34df09d83fadbdbbd6d7fce/vimeo-transcode-storage-prod-us-west1-h264-1080p/01/4691/27/698459270/3230977345.mp4`

**Comment by wukko** [comment]
@aquarime what? it works as expected, it gives you the video link

**Comment by aquarime** [comment]
@wukko , Cobalt does not download the Vimeo link, instead it opens it, i.e. the one mentioned above, on https://vod-progressive.akamaized.net/[...] as mentioned above.
YouTube videos are downloaded correctly, Dailymotion videos are downloaded correctly, not Vimeo videos.
I downloaded correctly the above mentioned Vimeo video with `https://9xbuddy.in/process?url=https%3A%2F%2Fvimeo.com%2F698459270` which means the Vimeo link is clean, not to mention viewable flawlessly on Vimeo itself.

**Comment by wukko** [comment]
@aquarime right click the video and save it. cobalt gives you the raw source file from vimeo's cdn. your browser previews it.

also, this is a pull request, not an issue

**Comment by aquarime** [comment]
@wukko,

> this is a pull request, not an issue

Indeed, sorry for that.

> right click the video and save it. cobalt gives you the raw source file from vimeo's cdn. your browser previews it.

OK, a new tab opened on `https://vod-progressive.akamaized.net/[...]`, had to shift+right_click to get the menu from which I downloaded the file.
Do agree that this is cumbersome compared to the way Cobalt handles other downloads with the traditional "Save file as" Windows' window. Being aware of this download oddity is required. I guess I'll handle Vimeo files with 9Xbuddy yet keep an eye on Cobalt given the amazing number of services it supports.

Thanks.

**Pull request merged**